app/src/main/java/com/asfoundation/wallet/wallet_validation/generic/PhoneValidationFragment.kt

```diff
@@ -99,17 +106,20 @@ class PhoneValidationFragment : DaggerFragment(),
     layout_validation_no_internet.visibility = View.GONE
   }
 
-  override fun getRetryButtonClicks(): Observable<Pair<String, String>> {
+  override fun getRetryButtonClicks(): Observable<PhoneValidationClickData> {
     return RxView.clicks(retry_button)
         .map {
-          Pair(ccp.selectedCountryCodeWithPlus,
-              ccp.fullNumber.substringAfter(ccp.selectedCountryCode))
+          PhoneValidationClickData(ccp.selectedCountryCodeWithPlus,
+              ccp.fullNumber.substringAfter(ccp.selectedCountryCode), previousContext ?: "")
         }
         .doOnNext { playRetryAnimation() }
         .delay(1, TimeUnit.SECONDS)
   }
 
-  override fun getLaterButtonClicks() = RxView.clicks(later_button)
+  override fun getLaterButtonClicks(): Observable<PhoneValidationClickData> {
+    return RxView.clicks(cancel_button)
+        .map { PhoneValidationClickData("", "", previousContext ?: "") }
+  }
```

Comment:
Couldn't it be interesting to know if the user already inserted the number?

Reply:
i don't think that's relevant. What matters is the intent, which in this case was to leave. If he tries and fails before, we'll know from other events